docs/index.rst

```diff
@@ -1,14 +1,162 @@
+       $ pip install -e .
+
+   .. note:: It is always recommended to use python virtual environment
+
+.. _usage_examples:
+
+Usage Examples
+``````````````
+.. note::
+
+  For easy understanding of betelgeuse, this repository is already included with
+  `data_sample` folder.  This folder contains sample tests and xml results which
+  will help in setting up and testing betelgeuse for your project.  The sample
+  commands used below also use this data.
+
+.. _help_command:
+
+help command
+++++++++++++
+
+.. code-block:: bash
+
+    $ betelgeuse --help
+
+.. _test_case_command:
+
+test-case command
++++++++++++++++++
+Creates/Syncs test cases with polarion. This command performs the following
+steps:
+
+- Testimony is called to parse the test cases.
+- For each parsed test case, the following actions are performed:
+
+    - The requirement name is derived from the test module name.  If the test
+      module name is `test_login_example.py`, then the requirement name is
+      `Login Example`.
+    - Requirement is created.
+    - Test case is created and linked to the requirement.
+
+.. code-block:: bash
+
+    $ betelgeuse test-case --path data_sample/tests/ PROJECT_CLOUD
+
+    Creating test case test_login_1 for requirement: Login Example.
+    Linking test case test_login_1 to requirement: Login Example.
+    Fetching requirement Login Example.
+    Creating requirement Login Example.
+
+.. note::
+  
+  * `PROJECT_CLOUD` is the polarion project id and not the project name.  This
+    can be found in Polarion -> Settings (icon) -> Administration -> ID.
+  * `path` is the path of the folder which has the test cases source code.
+
+.. warning::
+
+   Are you not sure if you are using this command in a right way?  No
+   problem! The test-case command can be used with `--collect-only` option
+   which runs in a dry run mode and shows the changes it would have made
+   without actually making them:
+
+     .. code-block:: bash
+
+         $ betelgeuse test-case --path data_sample/tests/ PROJECT_CLOUD \
+         --collect-only
+
+         Creating test case test_login_1 for requirement: Login Example.
+         Linking test case test_login_1 to requirement: Login Example.
+
+.. _test_results_command:
+
+test-results command
+++++++++++++++++++++
+Gives a nice summary of test cases/results in the given junit xml file.
+
+.. code-block:: bash
+
+    $ betelgeuse test-results --path \
+    data_sample/results/sample-junit-result.xml
+
+    Passed: 1
+
+.. _test_run_command:
+
+test-run command
+++++++++++++++++
+Creates a test run in polarion using the information in the given junit xml
+file. This command performs the following steps:
+
+- Parses the junit xml file to read all the test cases and their run statuses.
+- Creates a new test run and updates it with all the parsed test case items and
+  their run statuses.
+
+.. code-block:: bash
+
+    $ betelgeuse test-run --path data_sample/results/sample-junit-result.xml \
```

Comment:
In order to Betelgeuse proper map the `@ID` with the jUnit result it is needed that both the test runner and Betelgeuse get called on the same directory. With that said by having an entry like the following on the jUnit XML report:

```xml
<testcase classname="data_sample.tests.test_login_example.LoginTestCase" file="data_sample/tests/test_login_example.py" line="421" name="test_login_1" time="694.768339396">...</testcase>
```

Betelgeuse will map the test method based on its classname and name by joining them with a dot: `data_sample.tests.test_login_example.LoginTestCase.test_login_1` then Betelgeuse will convert every test module path into a python import path for example `data_sample/tests/test_login_example.py` will become `data_sample.tests.test_login_example` and all test methods or functions will then be appended later on, for the method `test_login_1` inside the class `LoginTestCase` it will generate `data_sample.tests.test_login_example.LoginTestCase.test_login_1`. With the two information matching then it will check the `@ID` token of that test method or function and will query Polarion for the work item id that represents that ID, finally with the work_item_id information Betelgeuse will add the result for a Polarion test case (work item) into the test run.